src/main/java/com/epam/aidial/cfg/service/transfer/importer/AdapterImporter.java

```diff
@@ -38,40 +44,52 @@ public class AdapterImporter {
     public Collection<ImportComponent<Adapter>> importAdapters(Map<String, CoreModel> coreModels,
                                                                ConfigImportOptions importOptions,
                                                                boolean isPreview) {
-        if (MapUtils.isNotEmpty(coreModels)) {
-            Map<String, Adapter> adapterByEndpoint = new HashMap<>();
-            for (Adapter adapter : adapterService.getAll()) {
-                adapterByEndpoint.put(adapter.getBaseEndpoint(), adapter);
-            }
-            Set<String> coreEndpoints = coreModels.values()
-                    .stream()
-                    .filter(coreModel -> coreModel.getEndpoint() != null)
-                    .map(this::mapToAdapterBaseEndpoint)
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toSet());
-            boolean createAdapterIfAbsent = importOptions.createAdapterIfAbsent();
-            List<ImportComponent<Adapter>> result = new ArrayList<>();
-            for (String endpoint : coreEndpoints) {
-                if (!adapterByEndpoint.containsKey(endpoint)) {
-                    if (createAdapterIfAbsent) {
-                        Adapter adapter = new Adapter();
-                        adapter.setBaseEndpoint(endpoint);
-                        if (!isPreview) {
-                            adapter.setName(UUID.randomUUID().toString());
-                            adapterService.create(adapter);
-                        } else {
-                            adapter.setName("<will be defined during import>");
-                        }
-                        result.add(new ImportComponent<>(ImportAction.CREATE, adapter));
+        if (MapUtils.isEmpty(coreModels)) {
+            return Collections.emptyList();
+        }
+
+        Set<String> adapterBaseEndpointsFromCoreModels = getAdapterBaseEndpoints(coreModels);
+        if (CollectionUtils.isEmpty(adapterBaseEndpointsFromCoreModels)) {
+            return Collections.emptyList();
+        }
+
+        Set<String> existingAdapterBaseEndpoints = adapterService.getAll().stream()
+                .map(Adapter::getBaseEndpoint)
+                .collect(Collectors.toSet());
+        boolean createAdapterIfAbsent = importOptions.createAdapterIfAbsent();
+
+        List<ImportComponent<Adapter>> result = new ArrayList<>();
+
+        int i = 0;
+        for (String endpoint : adapterBaseEndpointsFromCoreModels) {
+            if (!existingAdapterBaseEndpoints.contains(endpoint)) {
+                if (createAdapterIfAbsent) {
```

Comment:
minor: reverse this condition and exit early with an exception, avoiding big nested if-cases